src/Native/System.Security.Cryptography.Native/pal_bio.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_types.h"
+
+#include <openssl/bio.h>
+
+/*
+Creates a new BIO instance.
+*/
+extern "C" BIO* BioCreateNew();
+
+/*
+Direct shim to BIO_new_file.
+*/
+extern "C" BIO* BioNewFile(const char* filename, const char* mode);
+
+/*
+Cleans up and deletes a BIO instance.
+
+Implemented by:
+1) Calling BIO_free
+
+No-op if a is null.
+The given BIO pointer is invalid after this call.
+*/
+extern "C" int32_t BioDestroy(BIO* a);
+
+/*
+Direct shim to BIO_gets.
+*/
+extern "C" int32_t BioGets(BIO* b, char* buf, int32_t size);
+
+/*
+Direct shim to BIO_read.
+*/
+extern "C" int32_t BioRead(BIO* b, void* buf, int32_t len);
+
+/*
+Direct shim to BIO_write.
+*/
+extern "C" int32_t BioWrite(BIO* b, const void* buf, int32_t len);
+
+/*
+Shims the BIO_get_mem_data to get the size of data available in the BIO.
+*/
```

Comment:
I've seen some with "Function: *functionName*" in the documentation, but this seems overly redundant and prone to getting out-of-date.